test/common/http/conn_manager_impl_test.cc

```diff
@@ -2835,6 +2835,136 @@ TEST_F(HttpConnectionManagerImplTest, RequestTimeoutIsDisarmedOnConnectionTermin
   filter_callbacks_.connection_.raiseEvent(Network::ConnectionEvent::RemoteClose);
 }
 
+TEST_F(HttpConnectionManagerImplTest, RequestHeaderTimeoutValidlyConfigured) {
+  request_headers_timeout_ = std::chrono::milliseconds(10);
+  setup(false, "");
+
+  EXPECT_CALL(*codec_, dispatch(_)).WillOnce(Invoke([&](Buffer::Instance&) -> Http::Status {
+    Event::MockTimer* request_header_timer = setUpTimer();
+    EXPECT_CALL(*request_header_timer, enableTimer(request_headers_timeout_, _));
+    EXPECT_CALL(*request_header_timer, disableTimer());
+
+    conn_manager_->newStream(response_encoder_);
+    return Http::okStatus();
+  }));
+
+  Buffer::OwnedImpl fake_input("1234");
```

Comment:
Should this input be a partial valid request to avoid potential for header parse error?

Reply:
Yep, this test is more realistic now.